standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -2844,15 +2814,10 @@ public SQLAllTableConstraints getAllTableConstraints(String catName, String dbNa
       return rawStore.getAllTableConstraints(catName, dbName, tblName);
     }
 
-    Table tbl = sharedCache.getTableFromCache(catName, dbName, tblName);
-    if (tbl == null) {
-      // The table containing the constraints is not yet loaded in cache
-      return rawStore.getAllTableConstraints(catName, dbName, tblName);
-    }
     SQLAllTableConstraints constraints = sharedCache.listCachedAllTableConstraints(catName, dbName, tblName);
 
-    // if any of the constraint value is missing then there might be the case of partial constraints are stored in cached.
-    // So fall back to raw store for correct values
+    /* If constraint value is missing then there might be the case that table is not stored in cached or
```

Comment:
Add a TODO that this check is inefficient as all calls will hit rawstore if even one of the constraint type is not set for the given table.

Reply:
done